src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py

```diff
@@ -193,21 +194,129 @@ def _parse_slurm_args(
     def generate_full_srun_command(
         self, slurm_args: Dict[str, Any], env_vars: Dict[str, str], cmd_args: Dict[str, str], extra_cmd_args: str
     ) -> str:
+        """
+        Generate the full srun command for running a job on SLURM.
+
+        Args:
+            slurm_args (Dict[str, Any]): A dictionary containing SLURM arguments.
+            env_vars (Dict[str, str]): A dictionary containing environment variables.
+            cmd_args (Dict[str, str]): A dictionary containing command arguments.
+            extra_cmd_args (str): Additional command arguments.
+
+        Returns:
+            str: The full srun command as a string.
+        """
         self._create_run_script(slurm_args, env_vars, cmd_args, extra_cmd_args)
 
+        output_path = os.path.join(os.path.abspath(cmd_args["output_path"]), "output_pretest-%j-%n-%t.txt")
+        error_path = os.path.join(os.path.abspath(cmd_args["output_path"]), "error_pretest-%j-%n-%t.txt")
+
+        commands = []
+
+        pre_test_value = cmd_args.get("pre_test")
+
+        if pre_test_value == "True":
+            pre_test_command = self._generate_pre_test_command(cmd_args, output_path, error_path)
+            commands.append(pre_test_command)
+            # Check if the keyword is found in the pre-test output
+            pre_test_check_command = self._generate_pre_test_check_command(cmd_args, output_path)
+            commands.append(pre_test_check_command)
+
+        # Construct the srun command with the specific formatting
         srun_command_parts = [
             "srun",
-            f"--mpi={self.slurm_system.mpi}",
-            f"{self.slurm_system.extra_srun_args if self.slurm_system.extra_srun_args else ''}",
-            "--export=ALL",
-            f"-o {slurm_args['output']}",
-            f"-e {slurm_args['error']}",
-            f"--container-image={slurm_args['image_path']}",
-            f"--container-mounts={slurm_args['container_mounts']}",
+            f"--mpi={self.slurm_system.mpi} \\",
```

Comment:
Do we really need to add "\\"? Since we are using "join," can we include it as part of the arguments on line 239?

Reply:
Sure, we can add it to join or here. Same thing.

But I wanted to enforce the "\" in the generated srun commands. So added to the string generation itself. But yeah I see your point.